tests/modules/layers/test_parallel_attention.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest
+import torch
+
+from tests.test_utils import assert_expected  # , init_weights_with_constant
+from torchmultimodal.modules.layers.parallel_attention import ParallelAttentionBlock
+
+
+@pytest.fixture(autouse=True)
+def random():
+    torch.manual_seed(2023)
+
+
+class TestParallelAttentionBlocks:
+    @pytest.fixture
+    def embedding_dim(self):
+        return 16
+
+    @pytest.fixture
+    def total_layers(self):
+        return 1
+
+    @pytest.fixture
+    def mqa_num_heads(self):
+        return 1
+
+    @pytest.fixture
+    def gqa_num_heads(self):
+        return 2
+
+    @pytest.fixture
+    def num_heads(self):
+        return 16
+
+    @pytest.fixture
+    def max_seq_len(self):
+        return 32
+
+    @pytest.fixture
+    def mha_parallel_attention(self, embedding_dim, num_heads, total_layers):
+        pab_mha = ParallelAttentionBlock(
+            emb_dimension=embedding_dim,
+            num_heads=num_heads,
+            use_group_query_attention=False,
+            num_layers=total_layers,
+            use_weight_init=True,
+        )
+        pab_mha.eval()
+        return pab_mha
+
+    @pytest.fixture
+    def gqa_parallel_attention(
+        self, embedding_dim, num_heads, total_layers, gqa_num_heads
+    ):
+        pab_gqa = ParallelAttentionBlock(
+            emb_dimension=embedding_dim,
+            num_heads=num_heads,
+            use_group_query_attention=True,
+            num_heads_group_query_attention=gqa_num_heads,
+            num_layers=total_layers,
+            use_weight_init=True,
+        )
+        pab_gqa.eval()
+        return pab_gqa
+
+    @pytest.fixture
+    def mqa_parallel_attention(
+        self, embedding_dim, num_heads, total_layers, mqa_num_heads
+    ):
+        pab_mqa = ParallelAttentionBlock(
+            emb_dimension=embedding_dim,
+            num_heads=num_heads,
+            use_group_query_attention=True,
+            num_heads_group_query_attention=mqa_num_heads,
+            num_layers=total_layers,
+            use_weight_init=True,
+        )
+        pab_mqa.eval()
+        return pab_mqa
+
+    def test_mha_parallel_attention(
+        self,
+        mha_parallel_attention,
+        num_heads,
+        embedding_dim,
+        max_seq_len,
+    ):
+        # confirm all K and V keys match Q (i.e. MHA)
+        assert_expected(num_heads, mha_parallel_attention.num_kv)
+        # confirm num Q matches num_heads
+        assert_expected(num_heads, mha_parallel_attention.num_heads)
+
+        # input_ones = torch.ones(dims, dtype=torch.float)
+
+        x = torch.randint(0, 256, (1, max_seq_len, embedding_dim))  # bs =1,
```

Comment:
Can probably just make this a fixture (since I think it's used in all test cases)